### ISSUE TITLE
Having a callback after each error that happened when `allErrors` is `true`

### ISSUE THREAD

**Issue by suricactus** [ajv-validator/ajv#646]
**What version of Ajv you are you using?**
 Version 4.11.7, can upgrade
**What problem do you want to solve?**
tl;dr; Having a callback after each error that happened when `allErrors` is `true`.
Imagine extremely big JSON to validate, I would like to have a callback on each error and decide what to do next: continue or cancel validation. 
**What do you think is the correct solution to problem?**
Having an option like `errorCallback`, which is function with single argument: an Ajv error. If function returns true - should continue validation, if returns false - should break validation and return the list of errors so far.
**Will you be able to implement it?**
With some assistance I think I would be able to do it.

**Comment by epoberezkin** [comment]
What’s the use case?

**Comment by suricactus** [comment]
I believe that JSON Schema checks two different things in the same time:
* the schema - if the JSON is structured the correct way that an application or API understands
* the data - it conforms some less crucial validations, for example `minimum` and `maximum`
```
// schema1.json
{
  "type": "object",
  "properties": {
    "relatives": { "type": "array", "items": { "type": "string" } },
    "age": { "type": "integer", "minimum": 18 } 
  }
}
```

If we pass:
```
// data1.json
{
  "relatives": true,
  "age": 20
}
```
This is structure error, I expect `relatives` to be an array. All other errors does not make sense, because the JSON is not structured the way it should be.
```
// data2.json
{
  "relatives": ["Vladimir", "Ivan"],
  "age": 13
}
```
In this second example, the structure is ok, but it does not validate the data. I would like to collect all errors that are connected with data validation, because the submitted data is wrong and the party which input the data should know it.

The main difference is who is responsible for not passing validation. If I have structure error, it should be the software which packaged and gave me this wrong data to validate. On the other hand if the structure is ok, but data validation does not pass, I have to blame the input - user, sensor etc.

Now imagine we have extremely big JSON, let's say tens or hundreds of megabytes. If I put `allErrors: true` option, it will validate the whole JSON. If I have a callback, I would cancel the schema validator as soon as I encounter structure error

**Comment by epoberezkin** [comment]
Ok then, it doesn't need to be a new option I think, it can be just a function hook passed to allErrors option.

**Comment by epoberezkin** [comment]
Also, if somebody implements it, it should only call the hook on the errors that without allErrors option would terminate the validation (I.e. not inside compound keywords like “if”, “not” etc.).

**Comment by epoberezkin** [comment]
Closing for the lack of interest. Please vote on the issue if you need it - can be re-opened.
